README.md

```diff
@@ -19,14 +19,188 @@ The cleaned must-gather can then be found in the `must-gather-output-cleaned` fo
+Let's say one of your eth interfaces has the mac address `52:54:00:5e:ee:c6` and was logged, then `must-gather-clean`  will guarantee that it will always be assigned the same obfuscated consistent identifier across all files in a must-gather. 
+That primarily helps our support and engineers to ensure we can still understand and reproduces challenges that you were facing without putting your classified information at risk.
+
+Let's look at another obfuscation type named `IP`. This type can be used to clean IP addresses, we support both IPv4 and IPv6 except for local interfaces (`127.0.0.1`, `0.0.0.0` and `::1`) that will be preserved.
+
+You can configure this along with the MAC obfuscator like this:
+
+```
+config:
+  obfuscate:
+  - type: MAC
+    replacementType: Consistent
+  - type: IP
+    replacementType: Consistent
+```
+
+On a line-by-line basis, this will always execute the MAC obfuscation first and then the IP obfuscator - we'll go through this behaviour in more detail in the `Chaining obfuscators and side effects` section below. 
+
+Another configuration flag that we support for each obfuscation type is the `target`. The target is useful when the confidential information can be found not only in the file content, but also in the folder or file names. 
+This can very frequently happen with IP addresses, for example through node names. You can control that independently for each type like this:
+
+```
+config:
+  obfuscate:
+  - type: MAC
+    replacementType: Consistent
+    target: FileContents
+  - type: IP
+    replacementType: Consistent
+    target: FilePath
+```
+
+As you can see, the MAC obfuscator would work on file content whereas the IP obfuscator would work only on FilePaths. There is a mixed target called `All`, that will obfuscate on both paths and contents. 
+The default if no target is specified is `FileContents`, it is thus always recommended using the IP obfuscator with `target: All` to not accidentally leak IP information through folder names.
+
+The third built-in type of obfuscation is `Domain`, let's take a look how this can be configured:
+
+```
+config:
+  obfuscate:
+  - type: Domain
+    domainNames:
+    - "rhcloud.com"
+    - "dev.rhcloud.com"
+```
+
+As you can see, this type must be customized by supplying domain names. 
+Kubernetes resources are defined along with their domain names (eg "apps.openshift.io/v1") and thus would be automatically recognized as such and obfuscated as a false-positive.
+We thus kindly ask the user to supply their confidential domain names manually through the configuration.
+
+The above definition will obfuscate `rhcloud.com` as `domain0000001` (consistent) or as `xxxxxxxxxxxxx` (static). 
+Note that this does not include subdomains, they would need to be separately obfuscated. 
+A domain name defined as `staging.rhcloud.com` would only be obfuscated as `staging.domain0000001`, thus, it makes sense to also include all common subdomains as done in the example with "dev".
```

Comment:
May be reword a bit to avoid 'makes sense' phrase.
Additionally, _done in the example with "dev"_. Is this referred to `dev.rhcloud.com`? If yes, you can mention the entire url to be specific.

Reply:
reworded